src/Core/src/Platform/Gtk/MauiDatePicker.cs

```diff
@@ -1,35 +1,443 @@
 ﻿using System;
 using System.Runtime.Serialization;
+using System.Linq;
+using Gtk;
+using Microsoft.Maui.Graphics;
+using Microsoft.Maui.Graphics.Platform.Gtk;
 
 namespace Microsoft.Maui.Platform
 {
-	public class MauiDatePicker : Gtk.Label
+#pragma warning disable 612
+	public class DateEventArgs : EventArgs
 	{
+		DateTime _date;
 
-		public MauiDatePicker() : base()
+		public DateTime Date
 		{
-			Format = string.Empty;
+			get
+			{
+				return _date;
+			}
+		}
+
+		public DateEventArgs(DateTime date)
+		{
+			_date = date;
+		}
+	}
+
+	public class GrabHelper
+	{
+		private static uint CURRENT_TIME = 0;
+
+		public static void GrabWindow(Window window)
+		{
+			window.GrabFocus();
+
+			Grab.Add(window);
+
+			Gdk.GrabStatus grabbed =
+				Gdk.Pointer.Grab(window.Window, true,
+				Gdk.EventMask.ButtonPressMask
+				| Gdk.EventMask.ButtonReleaseMask
+				| Gdk.EventMask.PointerMotionMask, null, null, CURRENT_TIME);
+
+			if (grabbed == Gdk.GrabStatus.Success)
+			{
+				grabbed = Gdk.Keyboard.Grab(window.Window, true, CURRENT_TIME);
+
+				if (grabbed != Gdk.GrabStatus.Success)
+				{
+					Grab.Remove(window);
+					window.Destroy();
+				}
+			}
+			else
+			{
+				Grab.Remove(window);
+				window.Destroy();
+			}
+		}
+
+		public static void RemoveGrab(Window window)
+		{
+			Grab.Remove(window);
+			Gdk.Pointer.Ungrab(CURRENT_TIME);
+			Gdk.Keyboard.Ungrab(CURRENT_TIME);
+		}
+	}
+
+	public partial class DatePickerWindow : Window
+	{
+		VBox _datebox;
+		RangeCalendar _calendar;
+		public delegate void DateEventHandler(object sender, DateEventArgs args);
+
+		public event DateEventHandler? OnDateTimeChanged;
+
+		public DatePickerWindow()
+			: base(WindowType.Popup)
+		{
+			Title = "DatePicker";
+			TypeHint = Gdk.WindowTypeHint.Desktop;
+			WindowPosition = WindowPosition.Mouse;
+			BorderWidth = 1;
+			Resizable = false;
+			Decorated = false;
+			DestroyWithParent = true;
+			SkipPagerHint = true;
+			SkipTaskbarHint = true;
+
+			_datebox = new VBox();
+			_datebox.Spacing = 6;
+			_datebox.BorderWidth = 3;
```

Comment:
@parhamsaremi is XamarinForms' DatePicker hardcoding these values above too?

Reply:
Yes: https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.GTK/Controls/DatePicker.cs#L128